### PR TITLE
Fixes miscalculation on toWei func when passing valid numbers in scientific notation

### PR DESCRIPTION
**Pull request by nikoferro**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

Fixes `toWei` function so it can convert correctly valid numbers in scientific notation.

See also MMS-309
**Screenshots/Recordings**
Screenshot 2022-10-27 at 18 19 42
Screenshot 2022-10-28 at 14 52 55

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/util/number/index.js [hunk 1]
```diff
@@ -363,6 +363,23 @@ export function isNumber(str) {
...
+ * @see https://262.ecma-international.org/5.1/#sec-9.8.1
+ */
+
+export const isNumberScientificNotationWhenString = (value) => {
+  if (typeof value !== 'number') {
+    return false;
+  }
+  // toLowerCase is needed since E is also valid
+  return value.toString().toLowerCase().includes('e');
+};
```
Comment (gantunesr) [inline app/util/number/index.js:381]: Can you also add some basic unit test to this method?

**Review by gantunesr — COMMENTED** [review]
Hey @nikoferro, this LGTM. Just left a small comment

File: app/util/number/index.js (hunk 1 above)
Comment (nikoferro) [reply app/util/number/index.js:381]: thanks for taking a look @gantunesr . just added tests for this method

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by gantunesr** [comment]
@nikoferro do you think this PR needs QA before merging or a spot check on the release build is enough?

**Comment by nikoferro** [comment]
> @nikoferro do you think this PR needs QA before merging or a spot check on the release build is enough?

@gantunesr thanks for looking into. the unit test coverage over the function affected (`toWei`) seems that it will cover all the cases, but if there is bandwidth enough to have it QA'ed i think it would be better, since the `toWei` function is used on different places through the app

**Comment by sethkfman** [comment]
A nudge for @cortisiko & @chrisleewilcox on this topic: https://github.com/MetaMask/metamask-mobile/pull/5169#issuecomment-1313890275

**Pull request merged**